app/_kong_plugins/header-cert-auth/index.md

```diff
@@ -36,6 +36,144 @@ search_aliases:
+identifying the correct Consumer.
+
+{:.info}
+> **Note**: Subject names refer to the certificate's Subject Alternative Names (SAN) or
+Common Name (CN). CN is only used if the SAN extension does not exist.
+
+You can create a Consumer mapping with either of the following:
+  * The [`/consumers/{consumer}/header-cert-auth` Admin API endpoint](/api/gateway/admin-ee/#/operations/create-plugin-for-consumer)
+  * [decK](/gateway/entities/consumer/#set-up-a-consumer) by specifying `header_cert_auth_credentials` in the configuration like the following:
+    
+    ```yaml
+    consumers:
+    - custom_id: my-consumer
+      username: {consumer}
+      header_cert_auth_credentials:
+      - id: bda09448-3b10-4da7-a83b-2a8ba6021f0c
+        subject_name: test@example.com
+    ```
+
+The following table describes how the mapping parameters work:
+
+| Form Parameter                            | Default | Description |
+| ---                                       | ---     | --- |
+| `id`<br>*required for declarative config* |  none   | UUID of the Consumer mapping. Required if adding mapping using declarative configuration, otherwise generated automatically by {{site.base_gateway}}'s Admin API. |
+| `subject_name`<br>*required*              |  none   | The Subject Alternative Name (SAN) or Common Name (CN) that should be mapped to `consumer` (in order of lookup). |
+| `ca_certificate`<br>*optional*            |  none   | **If using the Admin API:** UUID of the Certificate Authority (CA). <br><br> **If using declarative configuration:** Full PEM-encoded CA certificate. <br><br>The provided CA UUID or full certificate has to be verifiable by the issuing certificate authority for the mapping to succeed. This is to help distinguish multiple certificates with the same subject name that are issued under different CAs. <br><br>If empty, the subject name matches certificates issued by any CA under the corresponding `config.ca_certificates`. |
+
+### Matching behaviors
+
+After a client certificate has been verified as valid, the Consumer object is determined in the following order, unless `skip_consumer_lookup` is set to `true`:
+
+1. Manual mappings with `subject_name` matching the certificate's SAN or CN (in that order) and `ca_certificate = <issuing authority of the client certificate>`
+2. Manual mappings with `subject_name` matching the certificate's SAN or CN (in that order) and `ca_certificate = NULL`
+3. If `config.consumer_by` is not null, Consumer with `username` and/or `id` matching the certificate's SAN or CN (in that order)
+4. The `config.anonymous` Consumer (if set)
+
+{:.info}
+> **Note**: Matching stops as soon as the first successful match is found.
+
+{% include_cached /plugins/upstream-headers.md %}
+
+When `skip_consumer_lookup` is set to `true`, Consumer lookup is skipped and instead of appending aforementioned headers, the plugin appends the following two headers
```

Comment:
```suggestion
When `config.skip_consumer_lookup` is set to `true`, Consumer lookup is skipped and instead of appending aforementioned headers, the plugin appends the following two headers
```